fundingmanager.go

```diff
@@ -801,17 +755,89 @@ func (f *fundingManager) reservationCoordinator() {
 	}
 }
 
-// advanceFundingState will advance the channel fromt the markedOpen state to
-// the point where the channel is ready for operation. This includes sending
-// funding locked to the peer, adding the channel to the router graph, and
-// announcing the channel. The updateChan can be set non-nil to get
-// OpenStatusUpdates.
+// advanceFundingState will advance the channel through the steps after the
+// funding transaction is broadcasted, up until the point where the channel is
+// ready for operation. This includes waiting for the funding transaction to
+// confirm, sending funding locked to the peer, adding the channel to the
+// router graph, and announcing the channel. The updateChan can be set non-nil
+// to get OpenStatusUpdates.
 //
 // NOTE: This MUST be run as a goroutine.
 func (f *fundingManager) advanceFundingState(channel *channeldb.OpenChannel,
-	updateChan chan<- *lnrpc.OpenStatusUpdate) {
+	pendingChanID [32]byte, updateChan chan<- *lnrpc.OpenStatusUpdate) {
 	defer f.wg.Done()
 
+	// If the channel is still pending we must wait for the funding
+	// transaction to confirm.
+	if channel.IsPending {
+		shortChanID, err := f.waitForFundingWithTimeout(channel)
+		if err == ErrConfirmationTimeout {
+			// We'll get a timeout if the number of blocks mined
+			// since the channel was initiated reaches
+			// maxWaitNumBlocksFundingConf and we are not the
+			// channel initiator.
+			err := fmt.Errorf("timeout waiting for funding tx "+
+				"(%v) to confirm", channel.FundingOutpoint)
+			fndgLog.Warnf(err.Error())
+
+			ch := channel
+			localBalance := ch.LocalCommitment.LocalBalance.ToSatoshis()
+			closeInfo := &channeldb.ChannelCloseSummary{
+				ChainHash:               ch.ChainHash,
+				ChanPoint:               ch.FundingOutpoint,
+				RemotePub:               ch.IdentityPub,
+				Capacity:                ch.Capacity,
+				SettledBalance:          localBalance,
+				CloseType:               channeldb.FundingCanceled,
+				RemoteCurrentRevocation: ch.RemoteCurrentRevocation,
+				RemoteNextRevocation:    ch.RemoteNextRevocation,
+				LocalChanConfig:         ch.LocalChanCfg,
+			}
+
+			if err := ch.CloseChannel(closeInfo); err != nil {
+				fndgLog.Errorf("Failed closing channel "+
+					"%v: %v", ch.FundingOutpoint, err)
+				return
+			}
+
+			// When the peer comes online, we'll notify it that we
+			// are now considering the channel flow canceled.
+			f.wg.Add(1)
+			go func() {
+				defer f.wg.Done()
+
+				peerChan := make(chan lnpeer.Peer, 1)
+				f.cfg.NotifyWhenOnline(ch.IdentityPub, peerChan)
+
+				var peer lnpeer.Peer
+				select {
+				case peer = <-peerChan:
+				case <-f.quit:
+					return
+				}
+				f.failFundingFlow(peer, pendingChanID, err)
```

Comment:
That's correct. However, seems like a discussion for a follow-up PR. Made a TODO.

Reply:
How do we handle this on the receiving side? This can be interpreted as a signal to double-spend our funding output.

Reply:
We will get a generic funding error on the receiving side, since it is not a whitelisted error. We could change this to some error code we understand, to act accordingly.